lte/gateway/python/magma/pipelined/app/classifier.py

```diff
@@ -312,31 +358,50 @@ def _resume_tunnel_flows(self, precedence:int, i_teid:int,
 
-        # discard downlink Tunnel for LOCAL port
-        if not ue_ip_adr:
-            self.logger.error("ue_ip_address is None")
-            return
+        if ip_flow_dl.set_params:
+            self._resume_tunnel_ip_flow_dl(ip_flow_dl)
         else:
-            ip_match_out = get_ue_ip_match_args(ue_ip_adr, Direction.IN)
-            match = MagmaMatch(eth_type=get_eth_type(ue_ip_adr),
-                           in_port=self._uplink_port, **ip_match_out)
+            # discard downlink Tunnel for LOCAL port
+            if not ue_ip_adr:
+                self.logger.error("ue_ip_address is None")
+                return
+            else:
+                ip_match_out = get_ue_ip_match_args(ue_ip_adr, Direction.IN)
+                match = MagmaMatch(eth_type=get_eth_type(ue_ip_adr),
+                                   in_port=self._uplink_port, **ip_match_out)
 
-            flows.add_resubmit_next_service_flow(self._datapath, self.tbl_num, match,
-                                                 priority=priority + 1,
-                                                 reset_default_register=False,
-                                                 resubmit_table=self.next_table)
+                flows.add_resubmit_next_service_flow(self._datapath, self.tbl_num, match,
+                                                     priority=priority + 1,
+                                                     reset_default_register=False,
+                                                     resubmit_table=self.next_table)
 
-            # discard downlink Tunnel for mtr port
-            match = MagmaMatch(eth_type=get_eth_type(ue_ip_adr),
-                               in_port=self.config.mtr_port, **ip_match_out)
+                # discard downlink Tunnel for mtr port
+                match = MagmaMatch(eth_type=get_eth_type(ue_ip_adr),
+                                   in_port=self.config.mtr_port, **ip_match_out)
 
-            flows.add_resubmit_next_service_flow(self._datapath, self.tbl_num, match,
-                                                 priority=priority + 1,
-                                                 reset_default_register=False,
-                                                 resubmit_table=self.next_table)
+                flows.add_resubmit_next_service_flow(self._datapath, self.tbl_num, match,
+                                                     priority=priority + 1,
+                                                     reset_default_register=False,
+                                                     resubmit_table=self.next_table)
+
+    def _resume_tunnel_ip_flow_dl(self, ip_flow_dl:IPFlowDL):
+        priority = Utils.get_of_priority(ip_flow_dl.precedence)
+
+        match = self._match_ip_flow_dl(ip_flow_dl, self._uplink_port)
+        flows.add_resubmit_next_service_flow(self._datapath, self.tbl_num, match,
+                                             priority=priority+1,
```

Comment:
I don't like this magical +1, why are we doing higher pri here?

Reply:
As per OAI code, whenever gtp tunnel resumed, the flow should be priority +1.

Reply:
Let's define the priority levels explicitly as enums to avoid any confusion.

discard/resume rule priorities > dedicated rule  priorities > default bearer priority > time-limited packet drop priority > paging rule priority > default rules (not specific to subscriber).